plugins/org.polarsys.capella.cybersecurity.sirius.analysis/src/org/polarsys/capella/cybersecurity/sirius/analysis/CybersecurityServices.java

```diff
@@ -740,8 +746,15 @@ public boolean removeCustomizationIfNeeded(DSemanticDecorator view) {
     EObject target = view.getTarget();
     DDiagram diagram = CapellaServices.getService().getDiagramContainer(view);
 
-    if (diagram != null && target instanceof AbstractFunction) {
-      EList<FunctionalChain> functionalChains = ((AbstractFunction) target).getInvolvingFunctionalChains();
+    if (diagram != null ) {
+      EList<FunctionalChain> functionalChains = null;
+      if (target instanceof AbstractFunction || target instanceof FunctionalExchange) {
+       functionalChains = ((AbstractFunction) target).getInvolvingFunctionalChains();

```

Comment:
if (target instanceof AbstractFunction)  -> remove the  || target instanceof FunctionalExchange